chore: try to get CI running with FUSE

Frando opened the pull request.

Frando committed:
chore: try to get CI running with FUSE

dignifiedquire commented:
looks like the install worked, and these failures are real now (for macos and nix)

Frando commented:
Hmm, tests pass for me locally on Linux without failures. Interesting.

The actual error is not reported, I'll change this to report the actual error (and not just a dropped channel).

Frando committed:
fix(fuse): bubble up errors on session creation

Frando committed:
fix(fuse): join FUSE thread on unmount in tests

Frando committed:
chore(fuse): more explicit temp dir handling

Frando committed:
chore(ci): install fuse for clippy & check

Frando committed:
chore(fuse): cleanup

Frando commented:
So:
1) On Linux, I'm getting a `No such file or directory (os error 2)` at `let session = fuser::Session::new(fs, &mountpoint, &options[..])?;`. Which is .. weird, because `    let mountpoint_meta = std::fs::metadata(&mountpoint)?;` just a couple of lines above does not error.
2) On MacOS it fails to compile, due to `libc` constants having different types. Hum. I'll try to find out how other crates handle this.

Frando committed:
chore: clippy

Frando committed:
chore(ci): Try to make FUSE tests work on *nix

The "Configure FUSE" step is taken from:
https://github.com/awslabs/mountpoint-s3/blob/main/.github/workflows/tests.yml#L46

Frando committed:
chore(ci): fix workflow

Frando committed:
fix(fuse): use libc type for mode

Frando committed:
chore: fmt

Frando committed:
fix(fuse): cast mode to correct type

Frando committed:
fix: disable FUSE on windows

Frando committed:
fix(ci): try to make fuse work on cross compile

Frando committed:
fix: fuse is only for unix at the moment

Frando committed:
chore(ci): make cross verbose for debugging

Frando committed:
fix: cross config

Frando commented:
OK, so Linux now passes, yay :rocket: 
FUSE is disabled now on Windows so Windows passes again, too.
MacOS seems to run forever, let's see what the logs say once the job is aborted.
The cross-compilations failed, I just pushed another commit that might fix this, let's see.

Frando commented:
@dignifiedquire or others with MacOS, can you check if the tests pass for you locally or if they hang as well?

Frando committed:
fix(ci): try to make cross work with FUSE

Frando committed:
chore(ci): another try at FUSE cross compilation
